pkg/clustermesh/endpointslicesync/endpointslice_test.go

```diff
@@ -201,24 +181,22 @@ func Test_meshEndpointSlice_Reconcile(t *testing.T) {
 		svcStore.CacheStore().Add(svc1)
 		serviceInformer.refreshAllCluster(svc1)
 
-		queue := getControllerQueue(controller)
-		require.NoError(t, waitEmptyQueue(queue))
-
-		epList, err := getEndpointSlice(&fakeClient, svcName)
-		require.NoError(t, err)
-		require.Zero(t, len(epList.Items))
+		require.EventuallyWithT(t, func(c *assert.CollectT) {
```

Comment:
This would not work as expected (and similarly the other ones asserting 0 epslices), because at the beginning the number of epslices is already 0. I'd suggest to manually create an epslice, and assert that it gets deleted.

Reply:
Yep indeed, and I actually caught a bug writing the correct code for that... When a service get deleted on a remote cluster we can never reconcile this deletion back :(.  TLDR that's because in the service informer the List call no longer returns this deleted service so we cannot call the cleanup hook afterwards. Will work on a fix this weekend!